test/parallel/test-fs-cp.mjs

```diff
@@ -312,6 +312,17 @@ function nextdir() {
   );
 }
 
+// It must not throw error if attempt is made to copy to dest
+// directory with same prefix as src directory
+// regression test for https://github.com/nodejs/node/issues/54285
+{
+  const src = nextdir('prefix');
+  const dest = nextdir('prefix-a');
+  mkdirSync(src);
+  mkdirSync(dest);
+  cpSync(src, dest, { recursive: true });
+}
```

Comment:
Can you also test `dir/prefix` and `dir/prefix-a`?

Reply:
This covers that. `nextdir` creates subdir under `test/tmp/` directory

Reply:
Actually, your updated comment sparked me. There are more corner cases to look at.

Reply:
Added more tests. This should be solid now.